src/lightning/app/core/flow.py

```diff
@@ -643,6 +649,8 @@ def experimental_iterate(self, iterable: Iterable, run_once: bool = True, user_k
         # TODO: Find a better way. Investigated using __reduce__, but state change invalidate the cache.
         if not user_key:
             frame = cast(FrameType, inspect.currentframe()).f_back
+            if frame is None:
```

Comment:
I think the returned frame can't be None, because you are always calling this method "from a frame". So I would add
`assert frame is not None`

Reply:
reverted :)